### PR TITLE
Fix beta clippy issues

### PR DESCRIPTION
**Pull request by likebreath**
Address clippy 1.54.0-beta.1 issues: `field is never read`, `needless-borrow`, `needless-collect`.

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

**Comment by likebreath** [comment]
@russell-islam This should fix the GH action completely (reference https://github.com/likebreath/cloud-hypervisor/pull/4). Let's land this PR first, and we can rebase other PRs.

**Comment by russell-islam** [comment]
> @russell-islam This should fix the GH action completely (reference [likebreath#4](https://github.com/likebreath/cloud-hypervisor/pull/4)). Let's land this PR first, and we can rebase other PRs.

Sounds good. Thanks for the fixes. :)

**Review by russell-islam — APPROVED** [review]

**Review by sboeuf — APPROVED** [review]

**Comment by sboeuf** [comment]
Thanks for fixing these issues @likebreath

File: vmm/src/acpi.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ pub const ACPI_APIC_GENERIC_REDISTRIBUTOR: u8 = 14;
 #[cfg(target_arch = "aarch64")]
 pub const ACPI_APIC_GENERIC_TRANSLATOR: u8 = 15;
 
+#[allow(dead_code)]
```
Comment (sboeuf) [inline vmm/src/acpi.rs:37]: @likebreath 
Ah I realized after I merged this PR that you could have specified the `#[allow(dead_code)]` for the specific field that is not used.
Could you send another PR to fix that? This way it's clearer what is not being used by the code, rather than tagging the entire structure as not being used.
